test/hardware/TestStub/HardwaretestStubWithMonitor.c

```diff
@@ -28,6 +27,9 @@
 #include <stdio.h>
 #include <string.h>
 
+networkCredentials_t networkCredentials = {.ssid = "SSID", .password = "password"};
```

Comment:
to prevent people/me from accidentally pushing these credentials we could add an example_network_credentials.h here, make the user/developer copy it to network_credentials.h and add network_credentials.h to .gitignore

Reply:
We have done this in the past which didn't work. And the things were also pushed.
If we want do stuff like that, I would recommend to do it via a library where the credentials are stored in a .c file which is ignored via gitignore. Setting variables like this in a header file can lead to problems if the file is accidentally included more than once.

Reply:
can't follow that argument.... that configuration would be local to this single build target, ie. `HardwaretestStubWithMonitor.c`

Reply:
This leaves the fact, that you define Variables inside a Header File, which is a bad idea to my point of view, because it leads to linkage problems quite fast.

Reply:
These are not global Headers. They are Headers for the HTTP/MQTT library, which are the libraries that require the user to specify these variable.
But yes your approach would also work. The benefit of my approach would be that we assure that these variables are defined.

I added an Issue for this matter: #171